File: languages/dart/examples/ComplexData.dart
import 'User.dart';

class ComplexData {
  List<int>? raw_bytes;
  List<String> items = [];
  Map<String, int> config = {};
  User? owner;

  void process(String mode) {
    // ...
  }

  static ComplexData create() {
    return ComplexData();
  }
}

class Metadata {
  String key = "";
  String value = "";

  void log() {}
}

File: languages/kotlin/examples/ComplexData.kt
class ComplexData {
    private val raw_bytes: ByteArray = ByteArray(0)
    private val items: List<String> = emptyList()
    private val config: Map<String, Int> = emptyMap()
    private val owner: User = User()

    fun process(mode: String) {
        // ...
    }

    fun create(): ComplexData {
        return ComplexData()
    }

    private class Metadata {
        var key: String = ""
        var value: String = ""

        fun log() {}
    }
}
